lib/Sema/CSSimplify.cpp

```diff
@@ -6951,6 +6963,32 @@ ConstraintSystem::SolutionKind ConstraintSystem::simplifyFixConstraint(
     return matchTypes(type1, type2, matchKind, subflags, locator);
   }
 
+  case FixKind::AllowTupleTypeMismatch: {
+    auto lhs = dyn_cast<TupleType>(type1.getPointer());
```

Comment:
I'd suggest to add an assert to `AllowTupleTypeMismatch::create` to make sure that both types are tuples and use `type{1, 2}->castTo<TupleType>()` here instead.